dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataOwnersManager.kt

```diff
@@ -0,0 +1,171 @@
+     */
+    @Transactional
+    fun getDataOwnerFromCompany(companyId: String): CompanyDataOwnersEntity {
+        val dataOwnersOfCompany = dataOwnerRepository.findById(companyId).getOrElse {
+            throw ResourceNotFoundApiException(
+                "No data owners found",
+                "The companyId '$companyId' do not have any data owners.",
+            )
+        }
+        return dataOwnersOfCompany
+    }
+
+    /**
+     * Method to delete a data owner from a given company
+     * @param companyId the ID of the company to which the data owner is to be deleted from
+     * @param userId the ID of the user who is to be removed as a data owner
+     * @return an entity holding the data ownership relations for the given company
+     */
+    @Transactional
+    fun deleteDataOwnerFromCompany(companyId: String, userId: String): CompanyDataOwnersEntity {
+        checkIfCompanyIsValid(companyId)
+        if (dataOwnerRepository.existsById(companyId)) {
+            val dataOwnersForCompany = dataOwnerRepository.findById(companyId).get()
+            if (dataOwnersForCompany.dataOwners.contains(userId)) {
+                dataOwnersForCompany.dataOwners.remove(userId)
+            } else {
+                throw ResourceNotFoundApiException(
+                    "Data owner not found",
+                    "User with Id $userId has not been data owner of company $companyId",
+                )
+            }
+            return if (dataOwnersForCompany.dataOwners.isEmpty()) {
+                dataOwnerRepository.deleteById(companyId)
+                CompanyDataOwnersEntity(companyId, dataOwnersForCompany.dataOwners)
+            } else {
+                dataOwnerRepository.save(
+                    CompanyDataOwnersEntity(
+                        companyId = companyId,
+                        dataOwners = dataOwnersForCompany.dataOwners,
+                    ),
+                )
+            }
+        } else {
+            throw ResourceNotFoundApiException(
+                "No data owners found",
+                "The companyId '$companyId' does not have any data owners.",
+            )
+        }
+    }
+
+    /**
+     * Method to check whether a specified user is a data owner of a given company, which throws an exception if not
+     * @param companyId the ID of the company
+     * @param userId the ID of the user
+     */
+    @Transactional
+    fun checkUserCompanyCombinationForDataOwnership(companyId: String, userId: String) {
+        checkIfCompanyIsValid(companyId)
+        val failException = ResourceNotFoundApiException(
+            "User is not a data owner",
+            "The user with Id $userId is not a data owner of the company with Id $companyId.",
+        )
+        if (!dataOwnerRepository.existsById(companyId)) {
+            throw failException
+        } else if (!dataOwnerRepository.getReferenceById(companyId).dataOwners.contains(userId)) {
```

Comment:
So if the user exists nothing is returned here and if the user does not exist / is data owner then an error is thrown, correct?